### PR TITLE
Integrate with CircleCI to build artifacts in the future

### PR DESCRIPTION
**Pull request by zoff99**
can't hurt
also circleCI has ssh access
and keeps the artifacts for download

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/369)

### CODE REVIEW

**Comment by GrayHatter** [comment]
@zoff99 good morning, while I take a look at this can you squash all these commits down into a single commit and give it a descriptive commit message?

**Comment by zoff99** [comment]
i mostly use the webinterface. it would be easier if you quash them on merge. there is a button for that here in the webgui.

**Comment by GrayHatter** [comment]
I'm happy to do that for you, but that's not an option on this repo, because it'll change the sha of the commit. There's a button to the right that enables contributors to push to your branch. If you check the box, I'll squash them for you.

**Comment by zoff99** [comment]
yaaaarp. done

**Comment by GrayHatter** [comment]
done, and done

**Comment by GrayHatter** [comment]
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[circle.yml, line 10 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_J9eGt2imZEbUwnFBj:-K_J9eGt2imZEbUwnFBk:b-5fuphi) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L10)):*
> ```YAML
>     - sudo apt-get update
>     - sudo apt-get install clang
>     - sudo sudo apt-get install build-essential libtool autotools-dev automake checkinstall check git yasm
> ```

sudo twice

---

*[circle.yml, line 33 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_J9q3ebp6bOGeEuvCF:-K_J9q3ebp6bOGeEuvCG:b2dslbw) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L33)):*
> ```YAML
>     ### ------- get libsodium -------
> 
> ##### ------------ BUILD ------------
> ```

why is the build/compile section listed under dependencies?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
here's the thing if you want see:

https://circleci.com/gh/zoff99/c-toxcore/92

**Comment by iphydf** [comment]
Review status: 0 of 1 files reviewed at latest revision, 7 unresolved discussions.

---

*[circle.yml, line 3 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lh65vxZYInf019MKC:-K_Lh65wfbR0Du_nZxHk:b-1t8lzq) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L3)):*
> ```YAML
> machine:
>   timezone:
>     Europe/Vienna
> ```

Perhaps make this UTC or something?

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_LgwheAGjXw6eF7OYo:blu6w7j) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
> ```YAML
>     - sudo apt-get install libopus-dev libvpx-dev pkg-config
> 
> # ------------ so that test number 7 succeeds ---------------
> ```

What's test number 7? Does it have a name?

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_LgihGWn2r3rLc-cvM:b-609ib) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
> ```YAML
>     ### ------- get libsodium -------
>     - mkdir ~/libsodium
>     - cd ~/libsodium/ ; git clone https://github.com/jedisct1/libsodium.git
> ```

I'd suggest using a PPA: apt-add-repository ppa:chris-lea/libsodium && apt-get update && apt-get install libsodium-dev

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_Lgg-AM2NHM6t8Ogqa:b-g0dci4) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
> ```YAML
>     - cd ~/libsodium/ ; cd libsodium/ ; git checkout tags/1.0.11
>     - cd ~/libsodium/ ; cd libsodium/ ; ./autogen.sh
>     - cd ~/libsodium/ ; cd libsodium/ ; ./configure && make check
> ```

Why do you run libsodium's tests on every build?

---

*[circle.yml, line 44 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LhDMeyVKkW9je5ZcY:-K_LhDMeyVKkW9je5ZcZ:b48y533) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L44)):*
> ```YAML
> test:
>   pre:
>     - echo "dummy"
> ```

What happens without this? Perhaps add a comment to say why this exists.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Reviewed 1 of 1 files at r1.
Review status: 0 of 1 files reviewed at latest revision, 8 unresolved discussions.

---

*[circle.yml, line 3 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lh65vxZYInf019MKC:-K_LnPt8y8jtw2EsC2D-:bb7f2mf) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L3)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Perhaps make this UTC or something?
</blockquote></details>

done

---

*[circle.yml, line 10 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_J9eGt2imZEbUwnFBj:-K_Ln_YRmfkVr8tO38UM:b-7ql1td) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L10)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

sudo twice
</blockquote></details>

fixed

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_LoAi8EXYUrWY5iXeB:b-7ql1td) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

What's test number 7? Does it have a name?
</blockquote></details>

fixed

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_LoDXBElDv4vao7AlP:b-j106yf) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

I'd suggest using a PPA: apt-add-repository ppa:chris-lea/libsodium && apt-get update && apt-get install libsodium-dev
</blockquote></details>

actually i want to build from source. maybe better pin to a specific commit of libsodium?

---

*[circle.yml, line 26 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LoSybC6fu-Hy8RG5F:-K_LoSybC6fu-Hy8RG5G:byvtt78) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L26)):*
> ```YAML
>     - mkdir ~/libsodium
>     - cd ~/libsodium/ ; git clone https://github.com/jedisct1/libsodium.git
>     - cd ~/libsodium/ ; cd libsodium/ ; git checkout tags/1.0.11
> ```

actually already pinning to 1.0.11 of libsodium

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_LoXwGEa3s0K2lo0hE:bg5uogc) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Why do you run libsodium's tests on every build?
</blockquote></details>

why not? does not hurt

---

*[circle.yml, line 33 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_J9q3ebp6bOGeEuvCF:-K_Loa_s7IZyT54o9KTr:blg75fi) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L33)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

why is the build/compile section listed under dependencies?
</blockquote></details>

that's just my quirk i guess. could be changed

---

*[circle.yml, line 44 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LhDMeyVKkW9je5ZcY:-K_LohBEXpUUpJOwYmjy:bi9dayj) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L44)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

What happens without this? Perhaps add a comment to say why this exists.
</blockquote></details>

i have it there so i does not "automagically guess some tests"
can try to remove it

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by GrayHatter** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_LryurKtGKFdQQGd3s:bgvdezm) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

fixed
</blockquote></details>

why is 7 still here? 

Also, this seems to be related to the travis ci hacks we're using. This would likely be a great place to start building v4 and v6

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_LrhVv06BjxFHgn8WA:b-k8rmxn) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

why not? does not hurt
</blockquote></details>

maybe we should cache them instead. It makes builds take longer, and is painful if you're trying to iterate quickly

---

*[circle.yml, line 33 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_J9q3ebp6bOGeEuvCF:-K_LrPqGHvFxosoiM9t2:b-n0tjw8) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L33)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

that's just my quirk i guess. could be changed
</blockquote></details>

yes please

---

*[circle.yml, line 44 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LhDMeyVKkW9je5ZcY:-K_LrUdJWY39IknsH7MM:b388sm8) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L44)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

i have it there so i does not "automagically guess some tests"
can try to remove it
</blockquote></details>

what's `ARGS="-V"` if it's just verbose you can use `make VERBOSE=1 test` instead

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Review status: 0 of 1 files reviewed at latest revision, 7 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_LtepOdq90lmcoh7e-:b-n3eurb) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

why is 7 still here? 

Also, this seems to be related to the travis ci hacks we're using. This would likely be a great place to start building v4 and v6
</blockquote></details>

what do you mean?
test number 7 on make test laster needs this entry in /etc/hosts, or it will fail. thats what this line does. and what the comments says.

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_LtzWA4b-GWvWBah6i:b-896fix) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

actually i want to build from source. maybe better pin to a specific commit of libsodium?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Review status: 0 of 1 files reviewed at latest revision, 7 unresolved discussions.

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_LuwF_fDh4HiPNKTmA:bqs5yia) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

maybe we should cache them instead. It makes builds take longer, and is painful if you're trying to iterate quickly
</blockquote></details>

dont know how to do this now. it works fine.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: 0 of 1 files reviewed at latest revision, 5 unresolved discussions.

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_M6cUTdsaSCeO1QuZe:baijlwu) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Why do you want to build from source?

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_M6gYD9w3QW_nk3FXZ:b-dysp46) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

dont know how to do this now. it works fine.
</blockquote></details>

It takes time, thus it has a cost to everybody involved. Why do you want to run them?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by GrayHatter** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_NGZw_BOAVNEGcKJRW:b-kwyhbm) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

what do you mean?
test number 7 on make test laster needs this entry in /etc/hosts, or it will fail. thats what this line does. and what the comments says.

</blockquote></details>

I'm finding it hard to parse in it's current form. There needs to be a descriptive comment here so I don't have to think that hard in the future.

---

*[circle.yml, line 26 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LoSybC6fu-Hy8RG5F:-K_NGqxIVKIFGla2zPBG:b-u7hghh) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L26)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

actually already pinning to 1.0.11 of libsodium
</blockquote></details>

@zoff99  you need to okay this. If you just want to add one off comments to reviewable in the future, you can start them with `BTW` or `FYI` that way you're not blocking on those comments.

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_NH7aJfS0WcT3RFyn-:b-9r9gnr) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

It takes time, thus it has a cost to everybody involved. Why do you want to run them?
</blockquote></details>

Also, do you need to cd every time, or is that just something you do here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_O7l-flWFG2H_OycjG:bxx6qsw) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

I'm finding it hard to parse in it's current form. There needs to be a descriptive comment here so I don't have to think that hard in the future.
</blockquote></details>

sure. make a suggestion? i am not sure what else to write

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_O7xIaRPeh60_GfIxP:bp376kw) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Why do you want to build from source?
</blockquote></details>

so i can switch easily (for testing or otherwise) to a different commit of libsodium.
lets say a new version is out and or want to test with a PR

but if you want this changed you can do it. it won't be usable for me then

---

*[circle.yml, line 28 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_Lgg-9ETlPo0VZgLRW:-K_O8JxA9eoePXQstmvw:b-sfka4p) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L28)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

Also, do you need to cd every time, or is that just something you do here?
</blockquote></details>

circleCI always starts in the repo directory for every command. so you need to go to the correct directory every time.
on ldconfig it's not needed i guess. but the it looks kinda weird, no?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
:lgtm_strong:

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-:-K_O93P-ZsJGKVK7vwvY:b-jbsf8z)*

**Comment by zoff99** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
first time using revieables :) so give me some learing time

**Comment by nurupo** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[circle.yml, line 30 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_ONXD1e9yYTJTyXs5W:-K_ONXD1e9yYTJTyXs5X:bw1tye3) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L30)):*
> ```YAML
>     - cd ~/libsodium/ ; cd libsodium/ ; ./configure && make check
>     - cd ~/libsodium/ ; cd libsodium/ ; sudo bash -c "printf 'y\naa\n\n' | checkinstall --install --pkgname libsodium --pkgversion 1.0.0 --nodoc --deldesc=no --pkglicense='GPL2'"
>     - cd ~/libsodium/ ; cd libsodium/ ; sudo ldconfig
> ```

Why do you navigate into the same directory every single command, instead of just doing it once?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[circle.yml, line 30 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_ONXD1e9yYTJTyXs5W:-K_OOVdavZp5vkacsUin:b-8vy2ic) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L30)):*
<details><summary><i>Previously, nurupo wrote…</i></summary><blockquote>

Why do you navigate into the same directory every single command, instead of just doing it once?
</blockquote></details>

circleCI always starts in the repo directory for every command. so you need to go to the correct directory every time.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by nurupo** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[circle.yml, line 30 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_ONXD1e9yYTJTyXs5W:-K_OOvHSqvwJ0N1EIURj:b2nhroa) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L30)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

circleCI always starts in the repo directory for every command. so you need to go to the correct directory every time.
</blockquote></details>

Wow

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[circle.yml, line 25 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgihGWn2r3rLc-cvL:-K_P55BGQiAeeya-V9-x:b-964ejg) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L25)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

so i can switch easily (for testing or otherwise) to a different commit of libsodium.
lets say a new version is out and or want to test with a PR

but if you want this changed you can do it. it won't be usable for me then
</blockquote></details>

If there is a use case, we can consider supporting it, but I can't see one. When do you actually want to do this? Why is it different for libsodium and libc? I consider the crypto functions to be a given and the API to be roughly as stable as the C standard library. For pull request testing, I don't think adding a few minutes of build time and a bunch of build script complexity (ynaan, ldconfig) that we need to maintain is worth it. If you want to test with a PR, you can spend your 1 minute of adding back these lines to the script and pushing it instead of everybody spending 1.5 minutes (20% of the total build time) on every push waiting for libsodium to build and run its internal test suite.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_P6OwlALXPJQiJD9Qw:b-pgq8e5) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

sure. make a suggestion? i am not sure what else to write
</blockquote></details>

"Test command" is less useful than "so that test xxx works". Reading "Test command", I have no idea what this line is about. Just replacing "test 7" with "network_test" would have been very useful. I'm ok with "so that network_test works".

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[circle.yml, line 34 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7AhpeSaI7dpwqWdX:-K_P7AhpeSaI7dpwqWdY:beyt6kz) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L34)):*
> ```YAML
> 
> ##### ------------ BUILD ------------
>     - cmake -DWARNINGS=OFF .
> ```

Why should the warnings be disabled on CI? I think the CI should enable as much as possible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 38 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7RaHHkcrx5IhZpWE:-K_P7RaHHkcrx5IhZpWF:b-3jvd62) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L38)):*
> ```YAML
>     - make
>     - sudo make install
>     - sudo ldconfig -v 2>/dev/null | grep toxcore
> ```

Is this actually needed?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by GrayHatter** [comment]
I'm all for this. But per @zoff99's request, prototype early, than iterate.

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 13 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_LgwheAGjXw6eF7OYn:-K_RQdEhB9JbbuMra6j9:bg1aol3) ([raw file](https://github.com/toktok/c-toxcore/blob/aacefb5418da7172949f2ae2575226a240dc582a/circle.yml#L13)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

"Test command" is less useful than "so that test xxx works". Reading "Test command", I have no idea what this line is about. Just replacing "test 7" with "network_test" would have been very useful. I'm ok with "so that network_test works".
</blockquote></details>

```network_test requires that `localhost` resolves to ::1```
```


---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-:-K_RQVPR2W02dTTl9hYv:b5yokow)*
<!-- Sent from Reviewable.io -->

**Comment by zoff99** [comment]
Review status: 0 of 1 files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 34 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7AhpeSaI7dpwqWdX:-K_T1KOtwwhUs0FQG_yj:b-bsm3u5) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L34)):*
<details><summary><i>Previously, iphydf wrote…</i></summary><blockquote>

Why should the warnings be disabled on CI? I think the CI should enable as much as possible.
</blockquote></details>

you guys told me to do so :-)
see: https://github.com/TokTok/c-toxcore/issues/293

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by GrayHatter** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 34 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7AhpeSaI7dpwqWdX:-K_TQIvCOXiqcuz8Os-I:bzgwrq1) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L34)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

you guys told me to do so :-)
see: https://github.com/TokTok/c-toxcore/issues/293

</blockquote></details>

If we're going to use it by default for our CI, then the warnings need to be enabled. Sorry for the confusion, I think nurupo was under the impression that you weren't planning to set up a whole new CI for us, and it was for a personal build. In which case disabling the warnings would have been the correct choice.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by zoff99** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 34 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7AhpeSaI7dpwqWdX:-K_TS3rWDD7jUV834yrK:bqp6z8g) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L34)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

If we're going to use it by default for our CI, then the warnings need to be enabled. Sorry for the confusion, I think nurupo was under the impression that you weren't planning to set up a whole new CI for us, and it was for a personal build. In which case disabling the warnings would have been the correct choice.
</blockquote></details>

not trying to replace travis. just adding a new system to compile. it does not have to do everything.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[circle.yml, line 34 at r3](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-K_P7AhpeSaI7dpwqWdX:-K_UdS28wnLWWONShhRC:b-870mtw) ([raw file](https://github.com/toktok/c-toxcore/blob/a9c7d8ca07cf6cba75eab61dd4eb4a0b1007b698/circle.yml#L34)):*
<details><summary><i>Previously, zoff99 (Zoff) wrote…</i></summary><blockquote>

not trying to replace travis. just adding a new system to compile. it does not have to do everything.
</blockquote></details>

Enable as much as possible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369)*

**Comment by iphydf** [comment]
@zoff99 I'm ok with merging this as is. We can iterate on it later, but I'd like to enable circleci as a pull request requirement for all future builds (in fact I've already done so). Can you rebase this on the current master?

**Comment by iphydf** [comment]
Actually, I can rebase this if others are happy for me to merge this.

**Comment by GrayHatter** [comment]
:lgtm_strong:

per irc conversation, PETI

---

Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/369#-:-K_WhUrN3JUyqppHpqkW:bu56cev)*

**Comment by iphydf** [comment]
It appears you've unchecked the box that allows us to rebase your branch. You'll have to do it yourself then. Please squash and rebase on master.

**Comment by zoff99** [comment]
sorry my bad. i've checked the box again
